src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -58,25 +61,70 @@
 	private static final String PUT = "PUT";
 	private String canvasToken;
 	private String canvasURL;
+	
+	private String callType;
+	
+	private String ltiUrl;
+	private String ltiKey;
+	private String ltiSecret;
+	
 	ResourceBundle props = ResourceBundle.getBundle("sectiontool");
 
-	
 	public void init() throws ServletException {
 		M_log.debug(" Servlet init(): Called");
 	}
 	
-	protected void doGet(HttpServletRequest request,HttpServletResponse response){
+	public void fillContext(Context context, HttpServletRequest request) {
+		ViewToolContext vtc = (ViewToolContext)context;
+		vtc.put("variable", "happiness");
+		M_log.info("fillContext() called");	
+	}
+	
+	public void doGet(HttpServletRequest request,HttpServletResponse response){
 		M_log.debug("doGet: Called");
 		try {
-			canvasRestApiCall(request, response);
+			if(request.getServletPath().equals("/manager")){
+				
+				canvasRestApiCall(request, response);
+			}
+			else{
+				M_log.info("NOT MANAGER");
+				doRequest(request, response);
+				//doRequest will call fillContext()
+			}
 		}catch(Exception e) {
 			M_log.error("GET request has some exceptions",e);
 		}
 	}
 	
-	protected void doPost(HttpServletRequest request,HttpServletResponse response){
+	public void doPost(HttpServletRequest request,HttpServletResponse response){
 		M_log.debug("doPOST: Called");
 		try {
+			//Is this an LTI Call or a browser call?
+			if(request.getParameterMap().containsKey("oauth_consumer_key")){
+				for (Object e : request.getParameterMap().entrySet()) {
+			        Map.Entry<String, String[]> entry = (Map.Entry<String, String[]>) e;
+			        String name = entry.getKey();
+			        for (String value : entry.getValue()) {
+			            System.out.println(name + " = " + value);
+			        }
+			    }
+				Properties appExtSecureProperties = SectionUtilityToolFilter.appExtSecurePropertiesFile;
+				if(appExtSecureProperties!=null) {
+					ltiKey = appExtSecureProperties.getProperty(SectionUtilityToolFilter.PROPERTY_LTI_KEY);
+					ltiSecret = appExtSecureProperties.getProperty(SectionUtilityToolFilter.PROPERTY_LTI_SECRET);
+					ltiUrl = props.getString(SectionUtilityToolFilter.PROPERTY_LTI_URL);
```

Comment:
Why is this ltiUrl Added in the sectiontool.properties file?  I think this can be pulled from the request Object or it could send it null i think. If you decide on having it as properties then  this should come from properties file external to the WAR